src/Raven.Client/Documents/Identity/AsyncHiLoIdGenerator.cs

```diff
@@ -73,9 +73,9 @@ public RangeValue(long min, long max, string serverTag)
         /// <summary>
         /// Generates the document ID.
         /// </summary>
-        /// <param name="entity">The entity.</param>
+        /// <param name="collectionName">The collection name.</param>
         /// <returns></returns>
-        public virtual async Task<string> GenerateDocumentIdAsync(object entity)
```

Comment:
this is one of those classes that we are allowing to inherit from, we cannot change the contract due to backward compatibility concerns.

I do not know if this is solvable without introducing any breaking changes, which we cannot introduce (only in major - 8.0).

If you will change the signature, then it is a breaking change for all of those that inherited and overridden this method and if another method overload is introduced then it is a possible behavior change, because the original method might not be called for those that inherited and overridden the method, hence changing the behavior.

@BickelLukas Do you see a way to solve this?

Reply:
alright, i made the discussed changes. Let me know if there's anything else